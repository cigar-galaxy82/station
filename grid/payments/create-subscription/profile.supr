name = "payments/create-subscription"
version = "1.0.0"

"""
Create subscription
"""

usecase CreateSubscription unsafe {
  input {
    """
    Plan ID
    ID of the plan to create a subscription for.
    """
    planId! string!

    """
    Customer Identificator
    Unique customer identficator specific to payment system (e.g. e-mail)
    """
    customer! string!
  }

  result {
    """
    ID
    Unique identifier of the subscription.
    """
    id! string!

    """
    Status
    Status of the subscription.

    Possible values:
     - `pending` - The subscription has been created, but not yet approved.
     - `approved` - The subscription has been approved by the buyer.
     - `active` - The subscription is active.
     - `trialing` - The subscription is currently in its trial period.
     - `past_due` - The subscription failed to be paid.
     - `suspended` - The subscription has been suspended.
     - `unpaid` - The subscription failed to be paid, and all retry attempts have been exhausted.
     - `cancelled` - The subscription has been cancelled.
     - `expired` - The subscription has expired.
    """
    status! enum {
      pending,
      approved,
      active,
      trialing,
      past_due,
      suspended,
      cancelled,
      expired
    }
  }

  error {
    """
    Title
    A short, human-readable summary of the problem type.
    """
    title! string

    """
    Detail
    A human-readable explanation specific to this occurrence of the problem.
    """
    detail string
  }

  example Successful {
    input {
      planId = "PLAN-89101829211"
      customer = "joe@example.com"
    }

    result {
      id = "91273981273918273"
      status = "active"
    }
  }
}