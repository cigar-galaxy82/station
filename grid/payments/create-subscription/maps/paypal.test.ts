import { createSubscriptionTest } from './create-subscription';

createSubscriptionTest('paypal', 'joe@example.com');
