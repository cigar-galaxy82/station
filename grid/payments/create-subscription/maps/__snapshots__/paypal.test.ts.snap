// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`payments/create-subscription/paypal CreateSubscription when all inputs are correct creates a subscription for a product 1`] = `
Object {
  "id": "I-8UXFU61581GN",
  "status": "pending",
}
`;
